server/src/main/java/org/elasticsearch/transport/InboundHandler.java

```diff
@@ -361,15 +378,21 @@ private static void sendErrorResponse(String actionName, TransportChannel transp
 
     private void handlerResponseError(StreamInput stream, final TransportResponseHandler<?> handler) {
         Exception error;
+        boolean readSuccess = false;
         try {
             error = stream.readException();
+            readSuccess = true;
         } catch (Exception e) {
             error = new TransportSerializationException(
                 "Failed to deserialize exception response from stream for handler [" + handler + "]",
                 e
             );
         }
```

Comment:
Why not just `ignoreDeserializationErrors : error` inside the catch block? Seems I am missing the point here :slightly_smiling_face: